libs/model-assessment/src/lib/ModelAssessmentDashboard/Context/buildModelAssessmentContext.ts

```diff
@@ -39,25 +39,23 @@ export function buildInitialModelAssessmentContext(
     | ISingleClassLocalFeatureImportance
     | undefined = undefined;
   if (
-    props &&
-    props.modelExplanationData.precomputedExplanations &&
-    props.modelExplanationData.precomputedExplanations.localFeatureImportance &&
-    props.modelExplanationData.precomputedExplanations.localFeatureImportance
-      .scores
+    props.modelExplanationData?.[0]?.precomputedExplanations
+      ?.localFeatureImportance?.scores
   ) {
     localExplanations =
-      props.modelExplanationData.precomputedExplanations.localFeatureImportance;
+      props.modelExplanationData[0].precomputedExplanations
+        .localFeatureImportance;
   }
   const jointDataset = new JointDataset({
     dataset: props.dataset.features,
     localExplanations,
     metadata: modelMetadata,
-    predictedProbabilities: props.modelExplanationData.probabilityY,
-    predictedY: props.modelExplanationData.predictedY,
+    predictedProbabilities: props.dataset.probabilityY,
+    predictedY: props.dataset.predictedY,
     trueY: props.dataset.trueY
   });
   const globalProps = buildGlobalProperties(
-    props.modelExplanationData.precomputedExplanations
+    props.modelExplanationData?.[0]?.precomputedExplanations
```

Comment:
This `?.[0]?` pops up all over the place. Should it really be a list if we always take the first element? Not just referring to this instance but all such in several files.

Reply:
no it should not, I listed few things in this pr comment.
This one we need to build a picker for user choose from.
Don't want to make this pr too huge that's why I always pick first one here.